lib/internal/fs/linux_recursive_watcher.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const { EventEmitter } = require('events');
+const path = require('path');
+const { Symbol, ObjectKeys } = primordials;
+
+const kFSWatchStart = Symbol('kFSWatchStart');
+
+let internalSync;
+let internalPromises;
+
+function lazyLoadFsPromises() {
+  internalPromises ??= require('fs/promises');
+  return internalPromises;
+}
+
+function lazyLoadFsSync() {
+  internalSync ??= require('fs');
+  return internalSync;
+}
+
+async function traverse(dir, files = {}) {
+  const { stat, readdir } = lazyLoadFsPromises();
+
+  files[dir] = await stat(dir);
+
+  try {
+    const directoryFiles = await readdir(dir);
+
+    for (const file of directoryFiles) {
+      const f = path.join(dir, file);
+
+      try {
+        const stats = await stat(f);
```

Comment:
Likewise, this is probably too slow for large directory trees.

`Stats` objects are fairly heavy-weight objects. The module loader, for example, goes out of its way to avoid creating them.

Ping-ponging through the microtask queue and the thread pool also adds considerable delay. It probably won't scale beyond a few thousand files.

Reply:
Is there are better alternative/approach to get `stat` of a file/folder?

Reply:
https://nodejs.org/dist/latest-v19.x/docs/api/fs.html#class-fsdirent

Reply:
Thank you @bnoordhuis & @targos. I'm working on replacing map to not use `stat` objects as a value. Meanwhile, I'm removing the `draft` status from the pull request. All tests are passing, and want to do a broader test using the CI.